flocker/control/httpapi.py

```diff
@@ -185,6 +185,134 @@ def saved(ignored):
+
+        :param unicode dataset_id: The unique identifier of the dataset.  This
+            is a string giving a UUID (per RFC 4122).
+
+        :param unicode primary: The address of the node to which the dataset
+            will be moved.
+
+        :param int maximum_size: The maximum number of bytes the dataset will
+            be capable of storing.  This may be optional or required depending
+            on the dataset backend.
+
+        :param dict metadata: A small collection of unicode key/value pairs to
+            associate with the dataset.  These items are not interpreted.  They
+            are only stored and made available for later retrieval.  Use this
+            for things like human-friendly dataset naming, ownership
+            information, etc.
+
+        :return: A ``dict`` describing the dataset which has been added to the
+            cluster configuration or giving error information if this is not
+            possible.
+        """
+        # XXX Changing volume size and metadata via the API is beyond the scope
+        # of this issue. Decide how to react.
+        # if maximum_size is None:
+        #     raise NotImplemented()
+
+        # if metadata is None:
+        #     raise NotImplemented()
+
+        # Get the current configuration
+        # deployment = self.persistence_service.get()
+
+        # Find all manifestations of dataset
+        # manifestations = manifestations_from_deployment(deployment, dataset_id)
+
+        # if not manifestations:
+        #     raise NotFound(
+        #         'Dataset not found. '
+        #         'There are no manifestations of the requested dataset. '
+        #         'dataset_id: {}'.format(dataset_id)
+        #     )
+
+        # Lookup the node that has a primary Manifestation (if any)
+
+        # expected_manifestation = Manifestation(dataset=dataset, primary=True)
+        # primary_nodes = list(
+        #     node for node in deployment.nodes if node.hostname == primary
+        # )
+        # if len(primary_nodes) == 0:
+        #     # No primary node
+        #     # raise something here?
+        #     # Spec says not
+        #     # "ie, don't accept non-IP addresses but do accept IP addresses that aren't part of the cluster"
+              # Ah there's already a comment about that....
+
+              # XXX Check cluster state to determine if the given primary node
+              # actually exists.  If not, raise PRIMARY_NODE_NOT_FOUND.
+              # See FLOC-1278
+        #     primary_node = None
+        # else:
+
+        #     # New primary was found.
+        #     # There should only be one.
+
+        #     (primary_node,) = primary_nodes
+
+        # if expected_manifestation in primary_node.manifestations():
+
+        #     # No change needed. return early?
+        #     pass
+
+        # Now construct a new_deployment where the primary manifestation of the dataset is on the requested primary node
```

Comment:
I would say... return an error indicating this is an invalid configuration change.